libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/usersearch/MatrixUserProfile.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.matrix.api.usersearch
+
+import io.element.android.libraries.matrix.api.core.UserId
+
+data class MatrixUserProfile(
```

Comment:
Isn't this almost the same as `MatrixUser`? Do we need to have the 2 of them?

Reply:
Yes, the only difference is that `MatrixUser` has an `AvatarData` member instead of just the URL. But that difference means that `MatrixUser` sits in the `matrixui` library and depends on the design system, so can't really be pulled in to the `matrix` module.

For comparison when we're dealing with rooms the `matrix` module has `RoomMember` which contains these details and the extra room-specific info, and then later is converted into a `MatrixUser` when presented in the UI.

Having the slight duplication seemed like the best option for now, unless there's some nicer way to handle this that I haven't seen?

Reply:
Yeah, looks like MatrixUser being Parcelable isn't required. Dropping that in favour of a data class in the `matrix` module with extension funcs in the `matrixui` makes sense to me.

Are you happy to land this change as-is then I'll do the MatrixUser->MatrixUserProfile refactor in a separate PR later today?

Reply:
Sure, let's do that.